### PR TITLE
Tree-repair

### LINKED ISSUE #8884: Set component doesn't work in GUI2
Trying to configure the set component via the drop downs doesn't work:

image

Found in build ide2-enso-win-2024.1.1-nightly.2024.1.29.exe

### PR DESCRIPTION
**Pull request by kazcw**
### Pull Request Description

Before committing an edit, insert any parentheses-nodes needed for the concrete syntax to reflect tree structure.

Fixes #8884.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [ ] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [ ] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [ ] Unit tests have been written where possible.
  - [ ] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

**Comment by kazcw** [comment]
(Stacked on #8880).

File: app/gui2/src/util/ast/__tests__/abstractFileIo.test.ts [hunk 1]
```diff
@@ -20,4 +20,7 @@ test('full file IdMap round trip', () => {
   expect(Ast.tokenTreeWithIds(ast2), 'Print/parse preserves IDs').toStrictEqual(astTT)
   expect(Ast.tokenTreeWithIds(ast_), 'All node IDs come from IdMap').toStrictEqual(astTT)
   expect([...idMap.entries()].sort()).toStrictEqual([...idMapOriginal.entries()].sort())
+
+  // Parsed tree houldn't need any repair.
```
Comment (farmaazon) [inline app/gui2/src/util/ast/__tests__/abstractFileIo.test.ts]:
```suggestion
  // Parsed tree wouldn't need any repair.
```

**Review by farmaazon — APPROVED** [review]
Add some docs to `repair` method to explain how does the repairing work.

**Comment by kazcw** [comment]
Integrated into #8893.

**Pull request closed without merging**